apps/meteor/client/views/admin/apps/AppDetailsPage/tabs/AppReleases/AppReleases.tsx

```diff
@@ -1,24 +1,43 @@
 import type { App } from '@rocket.chat/core-typings';
 import { Accordion } from '@rocket.chat/fuselage';
+import { useEndpoint, useToastMessageDispatch, useTranslation } from '@rocket.chat/ui-contexts';
+import { useQuery } from '@tanstack/react-query';
 import type { ReactElement } from 'react';
 import React from 'react';
 
-import { useEndpointData } from '../../../../../../hooks/useEndpointData';
-import { AsyncStatePhase } from '../../../../../../lib/asyncState/AsyncStatePhase';
 import AccordionLoading from '../../../AccordionLoading';
 import AppReleasesItem from './AppReleasesItem';
 
 // TODO: replace useEndpointData
 const AppReleases = ({ id }: { id: App['id'] }): ReactElement => {
-	const result = useEndpointData('/apps/:id/versions', { keys: { id } });
+	const getVersions = useEndpoint('GET', '/apps/:id/versions', { id });
+	const dispatchToastMessage = useToastMessageDispatch();
+	const t = useTranslation();
+
+	const { data, isLoading, isFetched } = useQuery(
+		['apps', id, 'versions'],
+		async () => {
+			const { apps } = await getVersions();
+
+			if (apps.length === 0) {
+				dispatchToastMessage({ type: 'error', message: t('No_results_found') });
```

Comment:
Let the error handler deal with it:

```suggestion
				throw new Error(t('No_results_found'));
```

Usually is not a good idea to run side-effects inside a query function, specially code that mutates React state (in this case, `dispatchToastMessage` will mutate the state of a context provider up in the element tree).